airflow/contrib/operators/oracle_to_oracle_transfer.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.hooks.oracle_hook import OracleHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class OracleToOracleTransfer(BaseOperator):
+    """
+    Moves data from Oracle to Oracle.
+
+
+    :param oracle_source_conn_id: source Oracle connection.
+    :type oracle_source_conn_id: str
+    :param sql: SQL query to execute against the source Oracle database. (templated)
+    :type sql: str
+    :param sql_params: Parameters to use in sql query. (templated)
+    :type sql_params: str
+    :param oracle_destination_conn_id: destination Oracle connection.
+    :type oracle_destination_conn_id: str
+    :param destination_table: destination table to insert rows.
+    :type destination_table: str
+    :param rows_chunk: number of rows per chunk to commit.
+    :type rows_chunk: str
+    """
+
+    template_fields = ('source_sql', 'source_sql_params')
+    ui_color = '#e08c8c'
+
+    @apply_defaults
+    def __init__(
+            self,
+            oracle_destination_conn_id,
+            destination_table,
+            oracle_source_conn_id,
+            source_sql,
+            source_sql_params={},
+            rows_chunk=5000,
+            *args, **kwargs):
+        super(OracleToOracleTransfer, self).__init__(*args, **kwargs)
+        self.oracle_destination_conn_id = oracle_destination_conn_id
+        self.destination_table = destination_table
+        self.oracle_source_conn_id = oracle_source_conn_id
+        self.source_sql = source_sql
+        self.source_sql_params = source_sql_params
+        self.rows_chunk = rows_chunk
+
+    def _execute(self, src_hook, dest_hook, context):
+        with src_hook.get_conn() as src_conn:
+            cursor = src_conn.cursor()
+            self.log.info("Querying data from source: {0}".format(
+                self.oracle_source_conn_id))
+            cursor.execute(self.source_sql, self.source_sql_params)
```

Comment:
Hi @Fokko , both `source_sql` and `source_sql_params` are templated fields. The reason to use extra params is because in some DBAPI, special characters are only considered when passed using parameters.